Tribler/Core/Modules/restapi/channel_endpoint.py

```diff
@@ -0,0 +1,100 @@
+import json
+import time
+
+from twisted.web import http, resource
+
+from Tribler.Core.simpledefs import NTFY_CHANNELCAST
+from Tribler.community.allchannel.community import AllChannelCommunity
+
+
+VOTE_UNSUBSCRIBE = 0
+VOTE_SUBSCRIBE = 2
+
+
+class ChannelEndpoint(resource.Resource):
+    """
+    This endpoint is the root endpoint responsible for handing all requests regarding a channel in Tribler.
+    """
+
+    def __init__(self, session):
+        resource.Resource.__init__(self)
+        self.session = session
+
+    def getChild(self, cid, request):
+        return SpecificChannelEndpoint(self.session, cid)
+
+
+class SpecificChannelEndpoint(resource.Resource):
+    """
+    This class delegates specific requests (i.e. fetching torrents in a channel or subscribing)
+    to child endpoint handlers.
+    """
+
+    def __init__(self, session, cid):
+        resource.Resource.__init__(self)
+        self.session = session
+        self.cid = cid
```

Comment:
Here it is cid and in the specificchannelbaseendpoint is is bytes(cid.decode(hex)) ?